src/lucky/page_helpers/url_helpers.cr

```diff
@@ -0,0 +1,65 @@
+module Lucky::UrlHelpers
+  # Tests if the given path matches the current request path.
+  #
+  # ```
+  # # Let's say we are visiting https://example.com/shop/products?order=desc&page=1
+  # current_page?("/shop/checkout")
+  # # => false
+  # current_page?("/shop/products")
+  # # => true
+  # current_page?("/shop/products/")
+  # # => true
+  # current_page?("/shop/products?order=desc&page=1")
+  # # => false
+  # current_page?("/shop/products?order=desc&page=1", check_query_params: true)
+  # # => true
+  # current_page?("https://example.com/shop/products")
+  # # => true
+  # current_page?("https://example.com/shop/products?order=desc&page=1")
+  # # => false
+  # ```
+  def current_page?(
+    path : String,
+    check_query_params : Bool = false
+  )
+    return false unless {"GET", "HEAD"}.includes?(@context.request.method)
```

Comment:
I ran some benchmarks and as it turns out, `Tuple#includes?` is by far the fastest option. Even faster than `==`:

```
includes? with % array  37.67M ( 26.55ns) (± 4.11%)  64.0B/op  12.39× slower
  includes? with array  37.81M ( 26.45ns) (± 3.71%)  64.0B/op  12.34× slower
  includes? with tuple 466.68M (  2.14ns) (± 3.95%)   0.0B/op        fastest
 includes? with string  49.91M ( 20.04ns) (± 1.65%)   0.0B/op   9.35× slower
               == & == 458.88M (  2.18ns) (± 5.26%)   0.0B/op   1.02× slower
                 match  20.62M ( 48.49ns) (± 3.14%)  16.0B/op  22.63× slower
```